mk/test.mk

```
@@ -1,41 +1,32 @@
 UPDATE_GOLDEN_FILES ?=
 TEST_PKG_LIST ?= ./...
-
-REPORTS_DIR ?= ./build/reports
-COVERAGE_PROFILE_FILENAME := coverage.out
-COVERAGE_PROFILE := $(REPORTS_DIR)/$(COVERAGE_PROFILE_FILENAME)
-COVERAGE_REPORT_HTML := $(REPORTS_DIR)/coverage.html
+REPORTS_DIR ?= build/reports
 
 GINKGO_TEST_FLAGS += -p --keep-going \
 	--keep-separate-reports \
-	--junit-report results.xml
-
-ifneq ($(REPORTS_DIR), "")
-	GINKGO_TEST_FLAGS += --output-dir $(REPORTS_DIR)
-endif
+	--junit-report results.xml \
+	--output-dir $(REPORTS_DIR)
 
 GINKGO_UNIT_TEST_FLAGS ?= \
-	--skip-package ./test,./pkg/transparentproxy/istio/tools --race \
-	--cover --covermode atomic --coverpkg ./... --coverprofile $(COVERAGE_PROFILE_FILENAME)
+	--skip-package ./test,./pkg/transparentproxy/istio/tools --race
 
-GINKGO_TEST:=$(GENV) $(GINKGO) $(GOFLAGS) $(LD_FLAGS) $(GINKGO_TEST_FLAGS)
+TEST_ENV=KUBEBUILDER_ASSETS=$(KUBEBUILDER_ASSETS) TMPDIR=/tmp UPDATE_GOLDEN_FILES=$(UPDATE_GOLDEN_FILES) $(GOENV)
+GINKGO_TEST:=$(TEST_ENV_VARS) $(GINKGO) $(GOFLAGS) $(LD_FLAGS) $(GINKGO_TEST_FLAGS)
 
 .PHONY: test
-test: build/ebpf
+test: build/ebpf ## Dev: Run tests for all modules
 	# -race required CGO_ENABLED=1 https://go.dev/doc/articles/race_detector and https://github.com/golang/go/issues/27089
-	KUBEBUILDER_ASSETS=$(KUBEBUILDER_ASSETS) TMPDIR=/tmp UPDATE_GOLDEN_FILES=$(UPDATE_GOLDEN_FILES) $(EXTRA_GOENV) CGO_ENABLED=1 go test $(GOFLAGS) $(LD_FLAGS) -race $$(go list $(TEST_PKG_LIST) | grep -E -v "test/e2e" | grep -E -v "test/blackbox_network_tests" | grep -E -v "pkg/transparentproxy/istio/tools")
+	$(TEST_ENV) CGO_ENABLED=1 go test $(GOFLAGS) $(LD_FLAGS) -race $$(go list $(TEST_PKG_LIST) | grep -E -v "test/e2e" | grep -E -v "test/blackbox_network_tests" | grep -E -v "pkg/transparentproxy/istio/tools")
 
 .PHONY: test-with-reports
-test-with-reports: build/ebpf ${COVERAGE_PROFILE} ## Dev: Run tests for all modules
-	KUBEBUILDER_ASSETS=$(KUBEBUILDER_ASSETS) TMPDIR=/tmp UPDATE_GOLDEN_FILES=$(UPDATE_GOLDEN_FILES) $(GINKGO_TEST) $(GINKGO_UNIT_TEST_FLAGS) $(TEST_PKG_LIST)
-	$(MAKE) coverage
-
-${COVERAGE_PROFILE}:
-	mkdir -p "$(shell dirname "$(COVERAGE_PROFILE)")"
-
-.PHONY: coverage
-coverage: ${COVERAGE_PROFILE}
-	GOFLAGS='${GOFLAGS}' go tool cover -html="$(COVERAGE_PROFILE)" -o "$(COVERAGE_REPORT_HTML)"
+test-with-reports: build/ebpf ## Dev: Run tests with test reports
+	$(GINKGO_TEST) $(GINKGO_UNIT_TEST_FLAGS) $(TEST_PKG_LIST)
+
+.PHONY: test-with-coverage
+test-with-coverage: build/ebpf
+	mkdir -p $(REPORTS_DIR)
+	$(GINKGO_TEST) $(GINKGO_UNIT_TEST_FLAGS) --cover --covermode atomic --coverpkg ./... --coverprofile coverage.out $(TEST_PKG_LIST)
+	GOFLAGS='${GOFLAGS}' go tool cover -html=$(REPORTS_DIR)/coverage.out -o "$(REPORTS_DIR)/coverage.html"
```

Wdyt about going with order-only-prerequisite for making sure the `$(REPORTS_DIR)` exists?
```suggestion
.PHONY: test-with-coverage
test-with-coverage: build/ebpf | $(REPORTS_DIR)
	$(GINKGO_TEST) $(GINKGO_UNIT_TEST_FLAGS) --cover --covermode atomic --coverpkg ./... --coverprofile coverage.out $(TEST_PKG_LIST)
	GOFLAGS='${GOFLAGS}' go tool cover -html=$(REPORTS_DIR)/coverage.out -o "$(REPORTS_DIR)/coverage.html"

$(REPORTS_DIR):
	mkdir -p $@
```